### PR TITLE
feat: stop all station service

### PR DESCRIPTION
**Pull request by vinteo**
Closes: #84

### CODE REVIEW

File: custom_components/opensprinkler/__init__.py [hunk 1]
```diff
@@ -367,6 +367,14 @@ def state(self):
         return self._state
 
 
+class OpenSprinklerControllerEntity:
+    def stop(self):
+        """Stops all stations."""
+        for _, station in self._controller.stations.items():
```
Comment (travisghansen) [inline custom_components/opensprinkler/__init__.py:373]: I think we should use this instead: `controller.stop_all_stations()` Although maybe we should rename that method and/or the service. Basically it stops any running stations *and* any stations that are `waiting`.

File: custom_components/opensprinkler/__init__.py [hunk 2]
```diff
@@ -367,6 +367,14 @@ def state(self):
         return self._state
 
 
+class OpenSprinklerControllerEntity:
+    def stop(self):
```
Comment (travisghansen) [inline custom_components/opensprinkler/__init__.py:371]: I'd probably name this a little more explicitly. `stop_stations` and then if we implement the other method `reset_stations`?

Comment (travisghansen) [reply custom_components/opensprinkler/__init__.py:371]: I've just updated the lib to have both methods. I think we should have both services.

Comment (vinteo) [reply custom_components/opensprinkler/__init__.py:371]: I didn't realise there was that method. Is there a scenario where we would want to stop all stations but not the waiting? Maybe the reset should be enough?

Comment (travisghansen) [reply custom_components/opensprinkler/__init__.py:371]: Yeah, good question. I've been surprised by the general sense that folks are wanting to build some pretty power scheduling/etc logic outside the unit itself (I may be misreading the sentiment). My comment is mostly driven by that feeling.

In that same vein I'd suggest we also add `enable`/`disable` services for the controller and possible even `reboot`.

Comment (vinteo) [reply custom_components/opensprinkler/__init__.py:371]: The controller already has an enable/disable switch, I don't see why it needs a service. The reboot service we can add if it is requested, but I don't see the need to add things that potentially might never get used.

Comment (vinteo) [reply custom_components/opensprinkler/__init__.py:371]: For this PR I think we should just keep the one service

Comment (travisghansen) [reply custom_components/opensprinkler/__init__.py:371]: That's fair...haven't delved into the automations too deeply yet so I'm a bit blind on this stuff. Meaning, if switches can generally be toggled in automations and don't require services then yeah, agreed.

File: custom_components/opensprinkler/__init__.py [hunk 3]
```diff
@@ -367,6 +367,12 @@ def state(self):
...
+        """Stops all stations."""
+        return self._controller.stop_all_stations()
```
Comment (vinteo) [inline custom_components/opensprinkler/__init__.py:373]: @travisghansen changed to just call the "reset"

**Pull request merged**